internal/pkg/manifest/validate.go

```diff
@@ -588,6 +589,24 @@ func (s StaticSiteConfig) validate() error {
 }
 
 func (f FileUpload) validate() error {
+	if err := f.validateSource(f.Source); err != nil {
+		return err
+	}
+	return nil
+}
+
+// validateSource returns nil if Source is configured correctly.
+func (f FileUpload) validateSource(source string) error {
```

Comment:
do you think we can read `f.Source` directly instead of passing `source` as a param?
```go
func (f FileUpload) validateSource() error {
  if f.Source == "" {}
}
```